### PR TITLE
2132 better formulas

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
@Yuri05 Just changed for one Rate. Can you check to make sure this is what you mean?

File: src/Db/Dump/PKSimDB_dump.txt [hunk 1]
```diff
@@ -3922,7 +3922,7 @@ INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveEffluxSpeci
...
 INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveInfluxSpecificWithTransporterInSource_Hill','TM * kcat * (K_water_src * max(C_src;0))^alpha / (Km^alpha + (K_water_src * max(C_src;0))^alpha)','Amount per time');
 INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','ActiveInfluxSpecificWithTransporterInSource_MM','TM * kcat * K_water_src * C_src / (Km + K_water_src * C_src)','Amount per time');
-INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','BrainPgPToPlasma_Hill','Min(CP * f_int * V * kcat * (fu * max(C_pls;0))^alpha / ( Km^alpha + (fu * max(C_pls;0))^alpha) ; P_endothelial * SA * fu * C_pls) + CP * f_int * V * kcat * (K_water_int * max(C_int;0))^alpha / (Km^alpha + (K_water_int * max(C_int;0))^alpha)','Amount per time');
+INSERT INTO tab_calculation_method_rates VALUES('LinksCommon','BrainPgPToPlasma_Hill','Min(TM * kcat * (fu * max(C_pls;0))^alpha / ( Km^alpha + (fu * max(C_pls;0))^alpha) ; P_endothelial * SA * fu * C_pls) + TM * kcat * (K_water_int * max(C_int;0))^alpha / (Km^alpha + (K_water_int * max(C_int;0))^alpha)','Amount per time');
```
Comment (msevestre) [inline src/Db/Dump/PKSimDB_dump.txt:3925]: TM for Transporter Amount. This is what we use in another transports

File: src/Db/Dump/PKSimDB_dump.txt [hunk 2]
```diff
@@ -85780,7 +85780,7 @@ INSERT INTO tab_rate_generic_molecules VALUES('LinksCommon','ActiveInfluxSpecifi
 INSERT INTO tab_rate_generic_molecules VALUES('LinksCommon','ActiveInfluxSpecificWithTransporterInSource_MM',366,'DRUG','TM',1);
 INSERT INTO tab_rate_generic_molecules VALUES('LinksCommon','BrainPgPToPlasma_Hill',341,'DRUG','C_int',0);
 INSERT INTO tab_rate_generic_molecules VALUES('LinksCommon','BrainPgPToPlasma_Hill',351,'DRUG','C_pls',0);
-INSERT INTO tab_rate_generic_molecules VALUES('LinksCommon','BrainPgPToPlasma_Hill',366,'DRUG','CP',0);
+INSERT INTO tab_rate_generic_molecules VALUES('LinksCommon','BrainPgPToPlasma_Hill',366,'DRUG','TM',1);
```
Comment (msevestre) [inline src/Db/Dump/PKSimDB_dump.txt:85783]: Use amount here

File: src/PKSim.Assets/PKSimConstants.cs [hunk 3]
```diff
@@ -403,7 +403,7 @@ public static class Error
          public static string DistributionUnknown(string distribution) => $"Distribution '{distribution}' is unknown.";
          public const string NameIsRequired = "Name is required.";
          public const string MoleculeIsRequired = "Molecule is required.";
-         public const string CategoryIsRequired = "Category is required.";
+         public static readonly string CategoryIsRequired = $"{UI.ExpressionProfileCategory} is required.";
```
Comment (msevestre) [inline src/PKSim.Assets/PKSimConstants.cs:406]: something else to get a better results

File: src/Db/Dump/PKSimDB_dump.txt (hunk 1 above)
Comment (Yuri05) [reply src/Db/Dump/PKSimDB_dump.txt:3925]: yep, looks good

**Comment by msevestre** [comment]
@Yuri05 All right. Should be all there. I am going to run the batch files to make sure there is no obvious corruption

File: src/Db/Dump/PKSimDB_dump.txt [hunk 4]
```diff
@@ -86106,16 +86106,16 @@ INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_Whole
...
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4683,'DRUG','M_int_prp',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4688,'DRUG','M_int_prc',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4693,'DRUG','M_bc_prp',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4698,'DRUG','M_bc_prc',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4703,'DRUG','M_cell_prp',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4708,'DRUG','C_cell_prc',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4713,'DRUG','C_pls_prp',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4718,'DRUG','C_pls_prc',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4723,'DRUG','C_end_prp',1);
+INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4728,'DRUG','C_end_prc',1);
```
Comment (Yuri05) [inline src/Db/Dump/PKSimDB_dump.txt:86118]: Those entries are not required anymore and should be deleted
```
INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4708,'DRUG','C_cell_prc',1);
INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4713,'DRUG','C_pls_prp',1);
INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4718,'DRUG','C_pls_prc',1);
INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4723,'DRUG','C_end_prp',1);
INSERT INTO tab_rate_generic_molecules VALUES('Observer_PKSim_2Poren','OBS_WholeOrgan_liv',4728,'DRUG','C_end_prc',1);
```

**Comment by Yuri05** [comment]
@msevestre ok, apart from https://github.com/Open-Systems-Pharmacology/PK-Sim/pull/2157/files#r843794150 looks good for me. When this is fixed we can merge and then I would run the batch comparison before we fix the pgp issue

**Comment by msevestre** [comment]
ok merging.

**Pull request merged**